esmvaltool/cmorizers/obs/cmorize_obs_regen.py

```diff
@@ -0,0 +1,95 @@
+"""ESMValTool CMORizer for REGEN data.
+
+Tier
+    Tier 2: other freely-available dataset.
+
+Source
+    https://researchdata.ands.org.au/rainfall-estimates-gridded-v1-2019/1408744
+Last access
+    20200226
+
+Download and processing instructions
+    Download the following files:
+        REGEN_AllStns_{version}_[1950..2016].nc
+
+"""
+
+import logging
+from pathlib import Path
+
+import cf_units
+import iris
+
+from esmvalcore.preprocessor import monthly_statistics
+from . import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def _extract_variable(short_name, var, cfg, file_path, out_dir):
+    """Extract variable."""
+
+    raw_var = var.get('raw', short_name)
+    cube = iris.load_cube(file_path, utils.var_name_constraint(raw_var))
+
+    # Fix units
+    cmor_info = cfg['cmor_table'].get_variable(var['mip'], short_name)
+    utils._set_units(cube, var.get('raw_units', short_name))
```

Comment:
```suggestion
    cube.units = var.get('raw_units', short_name)
```

should do the job in this case, but please be aware that `var.get('raw_units', short_name)` returns the actual `short_name` (e.g. `tas`, `pr`, etc.) if `raw_units` is not present, which is probably not what you want (most likely this will raise an Exception in `iris`).

Reply:
Thanks, I give it a try.

Reply:
I think what you want is something like

```python
if 'raw_units' in var:
    cube.units = var['raw_units']
```